src/SassBinary.php

```diff
@@ -156,11 +167,27 @@ public function getBinaryName(): string
 
     private function buildBinaryFileName(string $os, bool $isWindows = false): string
     {
-        return 'dart-sass-'.self::VERSION.'-'.$os.($isWindows ? '.zip' : '.tar.gz');
+        return 'dart-sass-'.$this->getVersion().'-'.$os.($isWindows ? '.zip' : '.tar.gz');
+    }
+
+    private function getDefaultBinaryPath(string $version): string
+    {
+        return $this->binaryDownloadDir.'/dart-sass/'.$version.'/sass';
+    }
+
+    private function getVersion(): string
+    {
+        return $this->cachedVersion ??= $this->binaryVersion ?? $this->getLatestVersion();
     }
 
-    private function getDefaultBinaryPath(): string
+    private function getLatestVersion(): string
     {
-        return $this->binaryDownloadDir.'/dart-sass/sass';
+        try {
+            $response = $this->httpClient->request('GET', 'https://api.github.com/repos/sass/dart-sass/releases/latest');
+
+            return $response->toArray()['tag_name'] ?? throw new \Exception('Cannot get the latest version name from response JSON.');
+        } catch (\Throwable $e) {
+            throw new \Exception('Cannot determine latest Dart Sass CLI binary version. Please specify a version in the configuration.', previous: $e);
```

Comment:
we have no error handling for rate limits in here, in my tests i get a `403` from github and the resulting error message can be confusing without reading to the `caused by` part

```
 Symfonycasts\SassBundle\Tests\SassBuilderTest::testSassOptionQuiet
Exception: Cannot determine latest Dart Sass CLI binary version. Please specify a version in the configuration.

/sass-bundle/src/SassBinary.php:191
/sass-bundle/src/SassBinary.php:181
/sass-bundle/src/SassBinary.php:38
/sass-bundle/src/SassBuilder.php:89
/sass-bundle/tests/SassBuilderTest.php:165

Caused by
Symfony\Component\HttpClient\Exception\ClientException: HTTP/2 403  returned for "https://api.github.com/repos/sass/dart-sass/releases/latest".
```